projects/storefrontlib/src/cms-components/myaccount/update-email/update-email.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import {
+  AuthService,
+  GlobalMessageService,
+  GlobalMessageType,
+  RoutingService,
+  UserService,
+} from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { CustomFormValidators } from '../../../shared/utils';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class UpdateEmailService implements OnDestroy {
+  protected subscriptions = new Subscription();
+  protected newUid: string;
+
+  constructor(
+    protected routingService: RoutingService,
+    protected globalMessageService: GlobalMessageService,
+    protected userService: UserService,
+    protected authService: AuthService
+  ) {
+  }
+
+  form: FormGroup = new FormGroup(
+    {
+      email: new FormControl('', [Validators.required, CustomFormValidators.emailValidator]),
+      confirmEmail: new FormControl('', [Validators.required]),
+      password: new FormControl('', [Validators.required])
+    },
+    {
+    validators: CustomFormValidators.emailsMustMatch('email', 'confirmEmail')
+  });
+
+  getUpdateEmailResultLoading(): Observable<boolean> {
```

Comment:
I'd clean up the method names, to get a nicer API, i.e.:

`updateEmailService.reset()`
`updateEmailService.save()`